tee-worker/litentry/core/evm-dynamic-assertions/src/precompiles/parse_int.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020-2024 Trust Computing GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::{failure_precompile_output, precompiles::PrecompileResult, success_precompile_output};
+use ethabi::ethereum_types::U256;
+use std::vec::Vec;
+
+pub fn parse_int(input: Vec<u8>) -> PrecompileResult {
+	let decoded = match ethabi::decode(&[ethabi::ParamType::String], &input) {
+		Ok(d) => d,
+		Err(e) => {
+			log::debug!("Could not decode string {:?}, reason: {:?}", input, e);
+			return Ok(failure_precompile_output(ethabi::Token::Uint(Default::default())))
+		},
+	};
+
+	let string_value = decoded.get(0).and_then(|v| v.clone().into_string());
+
+	let value = match string_value {
+		Some(v) => match U256::from_dec_str(v.as_str()) {
+			Ok(n) => n,
+			Err(e) => {
+				log::debug!("Cannot parse string {:?} to U256, reason: {:?}", v, e);
+				return Ok(failure_precompile_output(ethabi::Token::Uint(Default::default())))
+			},
+		},
+		None => {
+			log::debug!("Could not decode input {:?}, reason: string value is invalid", input);
+			return Ok(failure_precompile_output(ethabi::Token::Uint(Default::default())))
+		},
+	};
+
+	Ok(success_precompile_output(ethabi::Token::Uint(value)))
+}
+
+#[cfg(test)]
+pub mod test {
+	use crate::{
+		failure_precompile_output, precompiles::parse_int::parse_int, success_precompile_output,
+	};
+	use ethabi::{encode, Token};
+
+	#[test]
+	fn test_parse_int() {
+		// given
+		let encoded = encode(&[Token::String("100000000000000001".into())]);
+
+		// when
+		let result = parse_int(encoded).unwrap();
+
+		// then
+		assert_eq!(success_precompile_output(Token::Uint(100000000000000001_u128.into())), result)
+	}
+
+	#[test]
+	fn test_parse_int_fail() {
+		// given
+		let encoded = encode(&[Token::String("123XYZ".to_string())]);
+
+		// when
+		let result = parse_int(encoded).unwrap();
+
+		// then
+		assert_eq!(failure_precompile_output(Token::Uint(Default::default())), result)
+	}
+}
+
+#[cfg(test)]
+pub mod integration_test {
+	use crate::{execute_smart_contract, prepare_function_call_input};
+	use ethabi::{decode, encode, ethereum_types::U256, ParamType, Token};
+
+	// tee-worker/litentry/core/assertion-build/src/dynamic/contracts/tests/ParseInt.sol
```

Comment:
I was thiniking about where to place these tests and I decided to put it in package where smart contracts are located simply becuase the goal of these tests is to prove that solidity libraries works as expected (precompiled called and result properly resulted) but that's fine too.